tests/test_coreir/test_coreir_mux.py

```diff
@@ -25,8 +25,8 @@ def test_coreir_mux_2x3():
 
 
 def test_coreir_mux_2xTuple():
-    A = Tuple(a=Bit, b=Bits[2])
-    B = Tuple(a=A, b=Bits[3])
+    A = Product.from_fields("anon", dict(a=Bit, b=Bits[2]))
```

Comment:
what is the first argument here (`"anon"`)? some may argue this is a slightly more annoying syntax, any plan to (or would it be easy to) add some sugar?

Reply:
Products currently require a name, normally you say something like:
```
class MyType(Product):
    ...fields...
```
The first argument is equivalent to the `MyType` in the subclass syntax.  CC @cdonovick, could we add a version of from_fields that has a name parameter of `None`? (and give it a default anonymous name, this comes up often in magma with inline type definitions, e.g. `"IO", m.Tuple(x=m.In(m.Bit), y=m.Out(m.Bit))`.

Reply:
Yes we could have anonymous version of `Product`.  Just required an update: of https://github.com/leonardt/hwtypes/blob/master/hwtypes/adt_meta.py#L136